OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java

```diff
@@ -148,15 +143,15 @@ public void keyTyped(java.awt.event.KeyEvent evt) {
     }// </editor-fold>                        
 
     private void rSyntaxTextAreaKeyReleased(java.awt.event.KeyEvent evt) {
-        dirty = true;
-        setModified(true);
+        getEditorState().setDirty(true);

```

Comment:
No need to use getter in this class. Just change it to editorState.setDirty(true). This applies to all "getEditorState()" in THIS class.